doc/source/ray-air/examples/tfx_tabular_train_to_serve.ipynb

```diff
@@ -0,0 +1,2059 @@
+    "BATCH_SIZE = 8\n",
+    "# Number of epoch. Adjust it based on how quickly you want the run to be.\n",
+    "EPOCH = 1\n",
+    "# Number of training workers.\n",
+    "# Adjust this accordingly based on the resources you have!\n",
+    "NUM_WORKERS = 2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "whPRbBNbIrIl"
+   },
+   "source": [
+    "# Training"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "W7QYTpxXIrIl"
+   },
+   "source": [
+    "Let's starting with defining a simple Keras model for the classification task."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "id": "MwhAeEOuhYbV"
+   },
+   "outputs": [],
+   "source": [
+    "import tensorflow as tf\n",
+    "\n",
+    "def build_model():\n",
+    "    model = tf.keras.models.Sequential()\n",
+    "    model.add(tf.keras.Input(shape=(INPUT_SIZE,)))\n",
+    "    model.add(tf.keras.layers.Dense(50, activation=\"relu\"))\n",
+    "    model.add(tf.keras.layers.Dense(1, activation=\"sigmoid\"))\n",
+    "    return model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "UVVji2YKADrh"
+   },
+   "source": [
+    "Now let's define the training loop. This code will be run on each training\n",
+    "worker in a distributed fashion. See more details [here](https://docs.ray.io/en/latest/train/train.html)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {
+    "id": "U5pdjIzoAGRd"
+   },
+   "outputs": [],
+   "source": [
+    "from ray import train\n",
+    "from ray.train.tensorflow import prepare_dataset_shard\n",
+    "\n",
+    "def train_loop_per_worker():\n",
+    "    dataset_shard = train.get_dataset_shard(\"train\")\n",
+    "\n",
+    "    strategy = tf.distribute.experimental.MultiWorkerMirroredStrategy()\n",
+    "    with strategy.scope():\n",
+    "        model = build_model()\n",
+    "        model.compile(\n",
+    "            loss=\"binary_crossentropy\",\n",
+    "            optimizer=\"adam\",\n",
+    "            metrics=[\"accuracy\"],\n",
+    "        )\n",
+    "\n",
+    "    for epoch in range(EPOCH):\n",
+    "        # This will make sure that the training workers will get their own\n",
+    "        # share of batch to work on.\n",
+    "        # See `ray.train.tensorflow.prepare_dataset_shard` for more information.\n",
+    "        tf_dataset = prepare_dataset_shard(\n",
+    "            dataset_shard.to_tf(\n",
+    "                label_column=LABEL,\n",
+    "                output_signature=(\n",
+    "                    tf.TensorSpec(shape=(BATCH_SIZE, INPUT_SIZE), dtype=tf.float32),\n",
+    "                    tf.TensorSpec(shape=(BATCH_SIZE,), dtype=tf.int64),\n",
+    "                ),\n",
+    "                batch_size=BATCH_SIZE,\n",
+    "                drop_last=True,\n",
+    "            )\n",
+    "        )\n",
+    "\n",
+    "        model.fit(tf_dataset)\n",
```

Comment:
should you use a keras callback here to report intermediate results?

Reply:
Ah good point. I think it's more meaningful for tuning setup. Actually do you want to expand this notebook to include tuning or you think this may make things too complicated and we should start a followup tutorial for the tuning part?